core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -178,8 +178,32 @@ void HIPInternal::initialize(hipStream_t stream) {
   if (was_finalized)
     Kokkos::abort("Calling HIP::initialize after HIP::finalize is illegal\n");
 
+    // Get the device ID. If this is ROCm 5.6 or later, we can query this from
+    // the provided stream and potentially use multiple GPU devices. For
+    // ROCm 5.5 or earlier, we must use the singleton device id and there are no
+    // checks possible for the device id matching the device the stream was
+    // created on.
```

Comment:
So with ROCm 5.5 and earlier, we just send everything on a single device? Is there a way to warn the user about it?

Reply:
I believe the answer is no, there is not way to query the device ID from the stream until rocm5.6, so we couldn't tell the user. But currently in develop the user is free to create and pass a stream on the incorrect device without warning.

We could have a `HIP(dev_id)` constructor that takes in a device id and then internally creates and manages a stream, but this is very different to how we do this in Cuda and I think there has been push back on this in the past. But that would work for all rocm we support in Kokkos.

Reply:
This will need to be documented in the [Known issues](https://kokkos.org/kokkos-core-wiki/known-issues.html)

Reply:
Is there a documentation pr yet?